src/rust/engine/graph/src/entry.rs

```diff
@@ -423,6 +423,25 @@ impl<N: Node> Entry<N> {
             None,
             previous_result,
           )
+        } else if !self.node.content().cacheable() {
+          // Notify waiters
+          for waiter in waiters {
+            let _ = waiter.send(
+              previous_result
+                .clone()
+                .expect("I expected a result")
```

Comment:
What we should do here is either use the previous result (which we retrieved from the Node's current "running" state, or use the current result (which comes in as a parameter to this method).

So this is related to the case below in the "If the new result does not match the previous result" if/else block. If both the previous and current results are `None`, it would be justifiable to panic (for the reason mentioned in the method doc). But we need to check both of them.